lib/core/versions/latest/RequestHandler.ts

```diff
@@ -177,8 +177,8 @@ export default class RequestHandler implements IRequestHandler {
       // We reach here it means there is a DID Document to return.
 
       // If DID is published, use the short-form DID; else use long-form DID in document.
-      const didStringToUseInDidDocument = published ? did.shortForm : did.longForm!;
-      const document = DocumentComposer.transformToExternalDocument(didState, didStringToUseInDidDocument, published);
+      const didStringToUseInDidDocument = shortOrLongFormDid;
+      const document = DocumentComposer.transformToExternalDocument(didState, didStringToUseInDidDocument, published, did.isShortForm);
```

Comment:
Design consideration, what if we do:
```suggestion
      const document = DocumentComposer.transformToExternalDocument(didState, did, published);
```
Since
1. `didStringToUseInDidDocument ` is no longer 100% true since we transform it to short form in some cases. and
2. You won't need to repeat long-form to short-form logic.

Reply:
Good point. I like it.